pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -609,6 +609,10 @@ func (v *VolumeOptions) addVolumeToSpec(spec *kapi.PodSpec, info *resource.Info,
 			}
 			spec.Volumes = append(spec.Volumes[:i], spec.Volumes[i+1:]...)
 			break
+		} else {
+			if opts.Overwrite {
```

Comment:
Some users misunderstood that `--overwrite` can overwrite the volume like:

~~~
$ oc volume dc/docker-registry --add --overwrite --name=test --type=emptyDir
$ oc volume dc/docker-registry --add --overwrite --name=pvc1 --type=persistentVolumeClaim --claim-name=pvc1
~~~

and use `pvc1`.  But actual result is still using `test` volume like:

~~~
        volumeMounts:
        - mountPath: /registry
          name: test
      ...
      volumes:
      - emptyDir: {}          
        name: test
      - name: pvc1
        persistentVolumeClaim:
          claimName: pvc1
~~~

As you said, it should not be negative error, since it didn't break anything but just users misreading. However, do you have any other idea to prevent this misunderstanding?